azurelinuxagent/ga/exthandlers.py

```diff
@@ -647,12 +648,13 @@ def copy_status_files(self, old_ext_handler_i):
     def set_operation(self, op):
         self.operation = op
 
-    def report_event(self, message="", is_success=True):
+    def report_event(self, message="", is_success=True, duration=0):
         version = self.ext_handler.properties.version
         add_event(name=self.ext_handler.name, version=version, message=message, 
-                  op=self.operation, is_success=is_success)
+                  op=self.operation, is_success=is_success, duration=duration)
 
     def download(self):
+        begin_utc = datetime.datetime.utcnow()
```

Comment:
For the EG data, I am not sure if this is sufficient. We may need to have start and end events for frame analysis. Could you confirm?

Reply:
EG uses the timestamp and the duration.

A start and end event would be necessary if the event never completes or is interrupted.  If that did happen you would never get the event.  If you had a start and end event you would be able to see a start without a end.  I didn't want to double the number of events sent quite yet, which is why I started with this.

Reply:
Ok, sounds good